gmso/formats/mol2.py

```diff
@@ -51,165 +51,88 @@ def from_mol2(filename, site_type="atom"):
-                # check for the end of file
-                break
             else:
-                # else, skip to next line
-                line = f.readline()
+                supported_rti[section](
+                    topology, sections[section], site_type="atom"
+                )
+
     topology.update_topology()
     # TODO: read in parameters to correct attribute as well. This can be saved in various rti sections.
     return topology
 
 
-def _load_top_sites(f, topology, site_type="atom"):
-    """Take a mol2 file section with the heading '<TRIPOS>ATOM' and save to the topology.sites attribute.
-
-    Parameters
-    ----------
-    f : file pointer
-        pointer file where the mol2 file is stored. The pointer must be at the head of the rti for that
-        `@<TRIPOS>ATOM` section.
-    topology : gmso.Topology
-        topology to save the site information to.
-    site_type : string ('atom' or 'lj'), default='atom'
-        tells the reader to consider the elements saved in the mol2 file, and
-        if the type is 'lj', to not try to identify the element of the site,
-        instead saving the site name.
-
-    Returns
-    -------
-    line : string
-         returns the last line of the `@<TRIPOS>ATOM` section, and this is where the file pointer (`f`)
-         will now point to.
-
-    Notes
-    -----
-    Will modify the topology in place with the relevant site information. Indices will be appended to any
-    current site information.
-
-    """
-    while True:
-        line = f.readline()
-        if _is_end_of_rti(line):
-            line = line.split()
-            position = [float(x) for x in line[2:5]] * u.Å
-            # TODO: make sure charges are also saved as a unyt value
-            # TODO: add validation for element names
-            if site_type == "lj":
-                element = None
-            elif element_by_symbol(line[5]):
-                element = element_by_symbol(line[5])
-            elif element_by_name(line[5]):
-                element = element_by_name(line[5])
-            else:
-                warnings.warn(
-                    "No element detected for site {} with index{}, consider manually adding the element to the topology".format(
-                        line[1], len(topology.sites) + 1
-                    )
-                )
-                element = None
-            try:
-                charge = float(line[8])
-            except IndexError:
-                warnings.warn(
-                    "No charges were detected for site {} with index {}".format(
-                        line[1], line[0]
-                    )
-                )
-                charge = None
-            atom = Atom(
-                name=line[1],
-                position=position.to("nm"),
-                charge=charge,
-                element=element,
-                residue_name=line[7],
-                residue_number=int(line[6]),
+def _parse_atom(top, section, site_type="atom"):
```

Comment:
Instead of passing around kwargs to every function that doesn't need this site_type argument, it might be better to have a function named _parse_lj_site(top, section) that is separate from the _parse_atom function. Then we can add that to the supported_rti as:

```python
        site_parser = _parse_lj if atom_style == "lj" else _parse_atom
        supported_rti = {
            "@<TRIPOS>ATOM": site_parser,
            "@<TRIPOS>BOND": _parse_bond,
            "@<TRIPOS>CRYSIN": _parse_box,
            "@<TRIPOS>FF_PBC": _parse_box,
        }
```

Reply:
added separate method to parse `lj`